subsys/net/lib/websocket/websocket.c

```diff
@@ -518,6 +518,46 @@ static int websocket_ioctl_vmeth(void *obj, unsigned int request, va_list args)
 	return 0;
 }
 
+#if !defined(CONFIG_NET_TEST)
+static int sendmsg_all(int sock, const struct msghdr *message, int flags)
+{
+	int ret, i;
+	size_t offset = 0;
+	size_t total_len = 0;
+
+	for (i = 0; i < message->msg_iovlen; i++) {
+		total_len += message->msg_iov[i].iov_len;
+	}
+
+	while (offset < total_len) {
+		ret = zsock_sendmsg(sock, message, 0);
```

Comment:
We need to pass the flags to `sendmsg()` call. The timeout will be checked internally in sendmsg implementation so we do not need to do it here.

Reply:
Right, fixed now.